### PR TITLE
WIP gn1: Switch to generic vGPU device

### PR DESCRIPTION
**Pull request by lyarwood**
/cc @vladikr 

Would appreciate your thoughts here,  WIP/Draft as this is untested and I'm not sure this actually provides a vGPU yet.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: instancetypes/gn/1/gn1.yaml [hunk 1]
```diff
@@ -24,4 +24,4 @@ metadata:
 spec:
   gpus:
     - name: "gpu1"
-      deviceName: "nvidia.com/A400"
+      deviceName: "nvidia.com/gpu"
```
Comment (fabiand) [inline instancetypes/gn/1/gn1.yaml:27]: I'd not switch to a generic one, instead ideally this is useable out of the box.

Can we get NVIDIA to tell us what the most relevant GPU for this scenario is and use the resource here?
Or the top 3.

Comment (fabiand) [reply instancetypes/gn/1/gn1.yaml:27]: @rthallisey do you happen to know who can help here?

Comment (lyarwood) [reply instancetypes/gn/1/gn1.yaml:27]: > I'd not switch to a generic one, instead ideally this is useable out of the box.

The current choice of A400 or any other specific device name makes this unusable out of the box for the vast majority of users even with the device plugin configured and GPUs in the cluster IMHO.

> Can we get NVIDIA to tell us what the most relevant GPU for this scenario is and use the resource here?
> Or the top 3.

I think we should provide an example kustomise layer on top of common-instancetypes that sets a specific Nvidia device but default to the generic default vgpu device, assuming this works with KubeVirt.

**Comment by lyarwood** [comment]
/close

The GN class is now deprecated ahead of removal.

**Pull request closed without merging**